### ISSUE TITLE
General Analytics Request for VAOS-Community Care Provider Selection Rollout

### ISSUE THREAD

**Issue by laurenernest** [department-of-veterans-affairs/va.gov-team#20155]
# Metrics Reporting Request

## Description
Who is this request for?

- @laurenernest 
    
- What team/project is this work for?

    - VAOS - Community Care Provider Selection initiative
    
- What type of analytics support are you looking for?
  - [x] one time metrics report (this is not a KPI dashboard, but a temporary or a one time report)
  - [x] metrics clarification (verification that the metrics you are looking at are being read accurately)
  - [ ] analytics discussion (please share if you're looking for a general conversation, a KPI setting meeting, etc. in the "additional comments" section)
    

## Timeframe for Analytics Requested
_Please provide the timeframe for the metrics you are requesting._
What timeframes are you looking to analyze (e.g. before vs. after dd/mm/yy)?
Main goal is to track the performance of the canary/staged rollout. Feature toggle was turned on in Production 2/17 so I'd like to at least have comparisons from Jan 2021.

## Analytics to Track Down
- Overall CC conversion
- Conversion when getting the new page
- Submission totals and breakdown (provider preference count, languages count)

## Which Product Questions are you trying to answer?
- How many users are getting the new version of the page?
- How is our MVP performing?

## Other Helpful Questions to think through
- How frequently do you need this data? Will be reviewing this data daily

## Link to Metrics
- This is what I've played around with https://analytics.google.com/analytics/web/template?uid=nLyC-io1TcmeJppdUE4v-g

## Additional Comments
Here are the steps for the full funnel. One page would only be part of the new version behind the feature toggle.

1. eventLabel = 'vaos-schedule-appointment-button-clicked'
2. www.va.gov/health-care/schedule-view-va-appointments/appointments/new-appointment/
3. eventLabel = 'vaos-community-care-path-started'
4. www.va.gov/health-care/schedule-view-va-appointments/appointments/new-appointment/request-date/
5. www.va.gov/health-care/schedule-view-va-appointments/appointments/new-appointment/community-care-preferences/
6. www.va.gov/health-care/schedule-view-va-appointments/appointments/new-appointment/community-care-language _**(new page step in funnel. user would only see this if they got the provider selection version of the page at the preferences step)**_
7. www.va.gov/health-care/schedule-view-va-appointments/appointments/new-appointment/reason-appointment/
8. www.va.gov/health-care/schedule-view-va-appointments/appointments/new-appointment/contact-info/
9. www.va.gov/health-care/schedule-view-va-appointments/appointments/new-appointment/review
10. eventLabel = 'vaos-community-care-submission'
11. eventLabel = 'vaos-community-care-submission-successful'

> Please leave the following blank

## Acceptance Criteria
- [ ] The data has been identified and documented in this ticket - items that we don't have access to are noted with _cannot be accessed_
- [ ] The metrics are shared back with users.

## Definition of Done
- [ ] All appropriate issue tagging is completed
- [ ] All AC completed

**Comment by JasonTheBearded** [comment]
Hey @laurenernest since the feature toggle was flipped on in Prod on 2/17, are you wanting a compare of 1/17 - 2/17 and then 2/17 - 3/17 or all of Jan 2021 compared to all of Feb 2021?

**Comment by laurenernest** [comment]
Ideally, I'd like something I can use to modify the dates as needed, but if that's not possible, then:
- 8/1/20-2/16/21 compared to 2/17 to date
- Jan 2021 compared to Feb 2021
- March to date

The new step in the funnel and the submission totals are the 2 main things I'm having trouble getting myself.

**Comment by JasonTheBearded** [comment]
Hey @bsmartin-ep, I've completed the provider selection funnel viz, would you be able to QA the graphs and make sure everything looks right? The new viz will be on the Funnel subpage of the VAOS KPI Dash. Thanks!

**Comment by bsmartin-ep** [comment]
@JasonTheBearded - as far as users goes, right on the :nose:...

![image](https://user-images.githubusercontent.com/62304138/112334094-8b4a9700-8c91-11eb-8130-6776a74e4dc7.png)

**Comment by JasonTheBearded** [comment]
Thanks @bsmartin-ep! 

@laurenernest The new community care provider funnel is complete and available on the VAOS KPI Funnel page, https://va-gov.domo.com/page/443732151, let me know if you have any questions about the funnel or if you'd like for me to set up some time for a walkthrough. 

Thanks for your patience while we worked through this!

**Comment by laurenernest** [comment]
The abandonment between steps 3 and 4 (3-Community care preferences, 4-Community care languages) looks huge, but I suspect (and hope) that's because it's only based on the URL.

The same URL is used for both versions of the page so if the user gets the new version (vaos-communtiy-care-provider-selection-page) then they would see the language page next, but if they get the old version (vaos-community-care-legacy-provider-page) they would see the reason for appointment page next. Is it possible to narrow down the preferences step to only occurrences of the new version of the page?

Thanks for putting this together!

**Comment by JasonTheBearded** [comment]
Thanks @laurenernest! That step in the funnel is indeed only looking at the URL. I will check with my team to see if we can narrow it down to just the new page version.

**Comment by JasonTheBearded** [comment]
Hey @laurenernest I updated the funnel to narrow down the community care preferences to just the new version of the page. Let me know if that works for you. Thanks!

**Comment by laurenernest** [comment]
That looks better. Thanks!

The only other item in this ticket that I couldn't find was how to get the total submissions broken down by the additional data (number of provider preferences and language). I tried looking for the filters from [#16248](https://github.com/department-of-veterans-affairs/va.gov-team/issues/16248#issuecomment-733285325) but couldn't get the right data.

**Comment by jonwehausen** [comment]
Hi @laurenernest , 

I wanted to jump in here as I haven't seen any data/events to-date that indicate we're collecting the provider language. 

That said, I would recommend we add a DL variable or separate event indicating the user's language selection. Looking at `vets-website`, it looks like we are currently firing an event for provider clicks [(`vaos-choose-provider-click`)](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx#L141) . At language selection, if we could implement the following event we can begin to track the language selection. 

```javascript 
recordEvent({ 
  event: `vaos-choose-provider-language` ,
  'vaos-provider-language': 'english' //dynamically populate with language selection
});
```


cc: @JasonTheBearded

**Comment by laurenernest** [comment]
@jonwehausen @JasonTheBearded We might've missed something in our tracking. [Tagged](https://github.com/department-of-veterans-affairs/va.gov-team/issues/16972#issuecomment-809713669) you both in the ticket where we added it.

I'm essentially trying to answer the question of how many CC submissions don't have English as the preference. Similar to how we look at submission totals by type of care.

**Comment by JasonTheBearded** [comment]
@laurenernest Thanks for this! While we work through the language issue, I added a submission breakdown of provider preferences to the dash. It's currently checking all of community care, but if it needs to just look at users who received the new page let me know and I will filter the data down. 

Thanks again for your help and your patience!

**Comment by laurenernest** [comment]
The table you have on the dashboard is great! If I wanted to look at breakdown by old vs new versions of the page in GA, could you leave a note here for what I would need to set up so I can reference later.

**Comment by jonwehausen** [comment]
Hi @laurenernest , 

I wanted to provide an update on your request to expose analytics for the community care provider language selection -- After taking a look at Google Tag Manager, it looks as though at the time the `'vaos-number-of-preferred-providers'` data layer variable was configured for data collection, `'vaos-community-care-preferred-language'` was not. 

I greatly apologize for the miss here -- We have went ahead and made the GTM change to begin collecting the preferred language on submission. For your awareness moving forward, whenever there's additional metadata being passed along with an `event`, that's when our team is needed to provision the custom data collection within GTM. To expand a bit further, GTM changes are not necessary for simply collecting the existence of an `event` prefixed with `vaos-` , however any additional metadata sent with those events requires a minor GTM change. 

I apologize again for the inconvenience here -- as mentioned we have now started collecting the preferred provider language and can begin exposing within the dashboard point forward.  

cc: @JasonTheBearded @bsmartin-ep @laurensimpson14 @joanneesteban
